backend/benefit/common/tests/test_utils.py

```diff
@@ -51,3 +59,72 @@ def test_days360_errors(date1, date2):
 )
 def test_date_range_overlap(start_1, end_1, start_2, end_2, expected_result):
     assert date_range_overlap(start_1, end_1, start_2, end_2) == expected_result
+
+
+def test_get_date_range_end_with_days360():
+
+    assert get_date_range_end_with_days360(date(2022, 1, 1), 0) is None
+    assert (
+        get_date_range_end_with_days360(date(2022, 1, 1), decimal.Decimal("0.01"))
+        is None
+    )
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("0.03")
+    ) == date(2022, 1, 1)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("0.92")
+    ) == date(2022, 1, 28)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("1")
+    ) == date(2022, 1, 31)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 1), decimal.Decimal("1.04")
+    ) == date(2022, 2, 1)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 20), decimal.Decimal("1.38")
+    ) == date(2022, 2, 28)
+    assert get_date_range_end_with_days360(
+        date(2022, 1, 30), decimal.Decimal("6.99")
+    ) == date(2022, 8, 30)
+    # special case
+    assert (
+        get_date_range_end_with_days360(date(2022, 2, 28), decimal.Decimal("0.04"))
+        is None
+    )
+
+    if os.getenv("DAYS_360_FULL_TEST") == "1":
+        # the nested loops test a huge number of combinations. Takes a long time to run,
+        # only run this while debugging.
+        for n_days in range(0, 400):
+            print(f"start {n_days}")
+            for initial_day_offset in range(0, 100):
+                for months in [decimal.Decimal(m) / 100 for m in range(0, 1200)]:
+                    start_date = date(2022, 1, 1) + relativedelta(
+                        days=initial_day_offset
+                    )
+                    end_date = get_date_range_end_with_days360(start_date, months)
+                    # print(start_date, end_date, months)
+                    if end_date is None:
+                        # we are adding less than one day
+                        if not (start_date.month == 2 and start_date.day in [28, 29]):
+                            # end of February produces interesting results
+                            assert months < (1 / 30)
```

Comment:
…👀…hmm…

Are these corner cases tested in the normally run tests? If not then maybe at least having them be tested normally even without `DAYS_360_FULL_TEST` being set would be good for test coverage.

Reply:
Added some tests for February and leap year